### ISSUE TITLE
Lst.RemoveAll doesn't work

### ISSUE THREAD

**Issue by SirUppyPancakes** [louthy/language-ext#312]
Using the latest stable NuGet package (2.1.28). The following simple example reproduces it:

```csharp
var test = List(1, 2, 3, 4, 5);
Console.WriteLine(string.Join(", ", test.RemoveAll(x => x % 2 == 0)));
// output: 1, 2, 3, 4, 5
```
Compared to the equivalent approach using LINQ:

```csharp
var test = List(1, 2, 3, 4, 5);
Console.WriteLine(string.Join(", ", new Lst<int>(test.Where(x => x % 2 != 0))));
// output: 1, 3, 5
```

**Comment by louthy** [comment]
This is fixed in `v2.1.29`.  Stupid bug, but there's at least a unit test for it now!

**Comment by SirUppyPancakes** [comment]
As a side note, I imagine it is a lot of work writing unit tests for a library this large, and even harder to get good coverage of every function/utility. Since this is a functional library, might I suggest considering something along the lines of QuickCheck? (there is a .NET port available too: https://github.com/fscheck/FsCheck)

Admittedly, I haven't had a chance to try it out it myself (I've been meaning to), but the tedious nature of writing unit tests often causes me to give up on writing them altogether, This seems to make the process less painful.
